docs/2023-eb-eessi-uk-workshop/easybuild-contributing.md

```diff
@@ -1 +1,439 @@
+still can if you really want to).
+
+There are a couple of nice side effects too, like not having any local branches
+to tidy up on once the pull request gets merged (since `--new-pr` created the
+branch only in a temporary directory).
+
+If many contributions are made via `--new-pr` it also simplifies the task
+of EasyBuild maintainers, since pull requests opened this way have a particular
+structure to them and thus are easier to digest because they look familiar.
+
+### Uploading test report
+
+After opening the pull request, we should also upload a test report to show that the installation is working.
+This is just as easy as creating the pull request.
+
+First make sure that the pre-installed software in the prepared environment
+is available, since the required dependencies for `eb-tutorial` are already
+installed there:
+
+```shell
+module use /easybuild/modules/all
+```
+
+You can verify which dependencies are still missing using `--from-pr` combined with `--missing`:
+
+```shell
+# change '1' to the ID of your own pull request (see output of --new-pr)
+$ eb --from-pr 123 --missing
+== temporary log file in case of crash /tmp/eb-ioi9ywm1/easybuild-e3v0xa1b.log
+Please enter password for encrypted keyring:
+== found valid index for /home/example/.local/easybuild/easyconfigs, so using it...
+
+1 out of 20 required modules missing:
+
+* eb-tutorial/1.0.1-GCC-11.2.0 (eb-tutorial-1.0.1-GCC-11.2.0.eb)
+```
+
+Uploading a test report boils down to combining `--from-pr` with `--upload-test-report`:
+
+```shell
+# change '1' to the ID of your own pull request (see output of --new-pr)
+$ eb --rebuild --from-pr 123 --upload-test-report
+Please enter password for encrypted keyring:
+...
+== processing EasyBuild easyconfig /tmp/eb-bnb1pv3n/files_pr65/e/eb-tutorial/eb-tutorial-1.0.1-GCC-11.2.0.eb
+== building and installing eb-tutorial/1.0.1-GCC-11.2.0...
+...
+== COMPLETED: Installation ended successfully (took 2 sec)
+...
+Adding comment to easybuild-easyconfigs issue #65: 'Test report by @ebtutorial
+**SUCCESS**
+Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
+example - Linux centos linux 8.5, x86_64, Intel(R) Core(TM) E5-2660 v4 @ 2.00GHz (broadwell), Python 3.6.8
+See https://gist.github.com/f7c74159c809029afd99e30e4d994ef1 for a full test report.'
+== Test report uploaded to https://gist.github.com/f7c74159c809029afd99e30e4d994ef1 and mentioned in a comment in easyconfigs PR#1
+```
+
+Note that we may need to use `--rebuild` here since `eb-tutorial` may already be installed.
+
+This results in a comment being added to the pull request:
+
+<div align="center"><img src="../../img/test_report_comment.png" alt="Test report comment" width="75%"/></div>
+
+The gist linked from this comment provides more detailed information:
+
+<div align="center"><img src="../../img/test_report_gist.png" alt="Test report gist" width="75%"/></div>
+
+---
+
+[*(back to overview page)*](index.md)
```

Comment:
Eek, I'm wrong, I just looked at the schedule and this no longer occurs at the end! I suspect this is not only wrong case. Will fix in another PR